### ISSUE TITLE
Explicit task invocation

### ISSUE THREAD

**Issue by adamralph** [adamralph/bau#31]
``` Gherkin
Scenario:
  Given bau is required
    And a non-default task
    And another non-default task which depends on the first non-default task
    And a default task which invokes the second non-default task
    And the tasks are executed
   When I execute the baufile
   Then three tasks are executed
    And the first non-default task is executed first
    And the second non-default task is executed second
    And the default task is executed third
```

This is to allow things like:

``` C#
var bau = Require<BauPack>();

bau.Task("task1")
    .Do(() =>
    {
        if ( ... )
        {
            bau.Invoke("task2");
        }
    })
.Task("task2")
    .Do(() => { ... }).
.Execute();
```

The implementation _should_ already be complete as a side effect of other work. This _should_ only require the addition of an acceptance test.

**Comment by adamralph** [comment]
taken by @pgermishuys

BTW - the acceptance test itself doesn't need a conditional, that was more just an example of why it might be used.

I guess the script generated in the acceptance test should have 3 tasks A, B and C. C Can invoke B and B can depend on A. The test then asserts that C was executed, then A, then B. Make sense?

**Comment by adamralph** [comment]
@pgermishuys apologies, the original syntax I wrote was nonsense. This would never work because the `bau` variable is effectively not declared before the expression is trying to use it

``` C#
var bau = Require<BauPack>()
.Task("task1")
    .Do(() =>
    {
        if ( ... )
        {
            bau.Invoke("task2);
        }
```

I found this out today whilst trying to implement https://github.com/bau-build/bau/issues/54. The problem is that there is a bug in the Roslyn CTP which meant that a cryptic stack overflow execption was being thrown in this case, instead of a compiler warning. See https://github.com/scriptcs/scriptcs/issues/691.

I've updated the syntax accordingly, and there is no longer any need to change `Bau.Execute` to return anything. It can and _should_ remain as `void` since eventually it will be deprecated anyway. See https://github.com/bau-build/bau/issues/62
